### ISSUE TITLE
Fix addEnhancedEcommerceImpressionContext picking up the wrong positional arguments

### ISSUE THREAD

**Issue by volderette** [snowplow/snowplow-gtm-custom-template#14]
I was having trouble with the addEnhancedEcommerceImpressionContext as it always was marked as invalid schema. I realized, that the Snowplow library picks up the wrong values of the Datalayer. This seems to be because of the positional arguments the SP library is requiring:

This is an impression datalayer example:

```
        id: 'U-30-Abo ePaper und M-Plus',
        name: 'U-30-Abo ePaper und M-Plus',
        price: '14.9900',
        brand: 'Mittelbayerische Zeitung',
        category: 'catalog/category/view/s/neukunde-digitale-angebote/id/24/',
        position: 1,
        list: 'Spezial Angebote'
```

This is what ends up in the Snowplow hit:

```
id:	U-30-Abo ePaper und M-Plus
name:	U-30-Abo ePaper und M-Plus
list:	Spezial Angebote
brand:	Mittelbayerische Zeitung
category:	catalog/category/view/s/neukunde-digitale-angebote/id/24/
variant:	1
position:	14
```

Position will end up as variant and part of the price will end up as position.

**Comment by volderette** [comment]
I fixed it for my use case. I've added all the fields that can be set regarding to the Snowplow context (https://github.com/snowplow/snowplow/wiki/2-Specific-event-tracking-with-the-Javascript-tracker#3142-addenhancedecommerceimpressioncontext)

```
  if (getType(obj.impressions) === 'array') {
    obj.impressions.forEach(i => tracker('addEnhancedEcommerceImpressionContext',
                                         i.id,
                 				         i.name,
                                         i.list,
                                         i.brand,
                                         i.category,
                                         i.variant,
                                         i.position,
                                         i.price,
				                         obj.currencyCode));
    action = 'view';
  }
```

**Comment by sahava** [comment]
Thanks! This is fixed in PR #13, so PR #15 can be closed.
